graylog2-server/src/main/java/org/graylog2/contentpacks/facades/EntityWithExcerptFacade.java

```diff
@@ -91,6 +95,13 @@ default Optional<NativeEntity<EntityClass>> findExisting(Entity entity, Map<Stri
      */
     void delete(EntityClass nativeEntity);
 
+    /**
+     * Return a unique ID - primarily for troubleshooting
+     * @param nativeEntity The native entity
+     * @return unique ID of the entity
+     */
+    String id(ExcerptClass nativeEntity);
```

Comment:
@patrickmann Adding the non-default `id` method is a backward-incompatible change to the interface. I am not sure if anyone is using the content pack APIs in a 3rd-party plugin, but we should at least add an entry to the `UPGRADING.md` file to let people know about it.

Reply:
Good point - I will document that. 
I do think we need it, otherwise the error log messages would be pretty useless in helping to pinpoint the cause of a problem.